### DISCUSSION TITLE
What is the difference between ''join([innerJoin(), innerJoin()])'' and ''join([innerJoin()]).join([innerJoin()]) ''?

### DISCUSSION THREAD

**Discussion by ddxl123** [simolus3/drift#2004] [Q&A] [answered]
What is the difference between `join([innerJoin(), innerJoin()])` and `join([innerJoin()]).join([innerJoin()])`?
```
final sonAndFather = innerJoin( fathers, fathers.sonId.equals( sons.id ));
final matherAndFather = innerJoin( mathers, mathers.fatherId.equals( fathers.id ));
select(sons).join([sonAndFather, matherAndFather]);
```
`MatherAndFather` does not use `sons` to compare with `mathers`, so is there a problem with the query? If I can't do this query, how do I want to achieve such a query?

**Comment by simolus3** [comment, marked as answer]
There is no difference. Calling `join` multiple times will add the joins as well.

> `MatherAndFather` does not use sons to compare with `mathers`, so is there a problem with the query?

No, this is completely fine! Not all joins have to reference the original table, doing it like this should work. However, there is an optimization opportunity if you don't need all tables in the end. If you only care about some tables in the result (for instance, if you only need `sons` and `mathers` and joined `fathers` for the sole purpose of doing the second join), you can set `includeColumns: false` on the join to disable including it in the results. Bug again, this is just a matter of optimization - your existing way of writing joins will work fine.
